src/minichain/__main__.py

```
@@ -0,0 +1,1302 @@
+
+
+def _infer_sender_nonce_from_data_dir(*, data_dir: Path, sender_address: str) -> int:
+    node = MiniChainNode(NodeConfig(data_dir=data_dir))
+    try:
+        node.start()
+        return node.chain_manager.state.get_account(sender_address).nonce
+    except Exception as exc:
+        raise ValueError(
+            "unable to infer sender nonce from local state while daemon is running; "
+            "pass --nonce explicitly"
+        ) from exc
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)